index.bs

```diff
@@ -701,17 +784,28 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 				1. If |r| is not "granted", then reject |p| with a "NotAllowedError" DOMException
 
-				1. Let |cleanItemList| be an empty sequence&lt;{{Blob}}>.
+				1. Let |itemList| and |cleanItemList| be an empty sequence&lt;{{Blob}}>.
+
+				1. For each |itemData| in |data|:
+
+					1. For each |item| in |itemData|:
+						1. If [=Promise=] to {{Blob}} in |item| is rejected, then throw "Promises to Blobs were rejected." DOMException.
+
+						1. Add the {{Blob}} in |item| to |itemList| after the promise has been resolved.
+
+					1. For each {{Blob}} |itemBlob| in |itemList|:
+
+						1. If {{Blob/type}} is not in the [=mandatory data types=] list, then reject |p| with a "Type {{Blob/type}} not supported on write." DOMException.
```

Comment:
`type` needs to be defined from somewhere. I guess it would have to be stored corresponding to the blobs in `itemList`.

Reply:
Done.